lib/model.go

```diff
@@ -124,3 +124,62 @@ type RenameTopicRequest struct {
 type SubscribersTopicRequest struct {
 	Subscribers []string `json:"subscribers"`
 }
+
+type IntegrationCredentials struct {
+	ApiKey           string                 `json:"apiKey,omitempty"`
+	User             string                 `json:"user,omitempty"`
+	SecretKey        string                 `json:"secretKey,omitempty"`
+	Domain           string                 `json:"domain,omitempty"`
+	Password         string                 `json:"password,omitempty"`
+	Host             string                 `json:"host,omitempty"`
+	Port             string                 `json:"port,omitempty"`
+	Secure           bool                   `json:"secure,omitempty"`
+	Region           string                 `json:"region,omitempty"`
+	AccountSid       string                 `json:"accountSid,omitempty"`
+	MessageProfileId string                 `json:"messageProfileId,omitempty"`
+	Token            string                 `json:"token,omitempty"`
+	From             string                 `json:"from,omitempty"`
+	SenderName       string                 `json:"senderName,omitempty"`
+	ProjectName      string                 `json:"projectName,omitempty"`
+	ApplicationId    string                 `json:"applicationId,omitempty"`
+	ClientId         string                 `json:"clientId,omitempty"`
+	RequireTls       bool                   `json:"requireTls,omitempty"`
+	IgnoreTls        bool                   `json:"ignoreTls,omitempty"`
+	TlsOptions       map[string]interface{} `json:"tlsOptions,omitempty"`
+}
+
+type CreateIntegrationRequest struct {
+	ProviderId  string                 `json:"providerId"`
+	Channel     ChannelType            `json:"channel"`
+	Credentials IntegrationCredentials `json:"credentials,omitempty"`
+	Active      bool                   `json:"active"`
+	Check       bool                   `json:"check"`
+}
+
+type UpdateIntegrationRequest struct {
+	Credentials IntegrationCredentials `json:"credentials"`
+	Active      bool                   `json:"active"`
+	Check       bool                   `json:"check"`
+}
+
+type Integration struct {
+	Id             string                 `json:"_id"`
+	EnvironmentId  string                 `json:"_environmentId"`
+	OrganizationId string                 `json:"_organizationId"`
+	ProviderId     string                 `json:"providerId"`
```

Comment:
Update `EnvironmentID`, `OrganizationID` and `ProviderID`

Reply:
Hey @samsoft00 , actually other previous structs like `GetTopicResponse` have the same naming form. To maintain consistency, I used the same. 
Should I change it for `GetTopicResponse` also ?

Reply:
Nope, only update the keys as 

```go
type Integration struct {
	Id             string                 `json:"_id"`
	EnvironmentID  string                 `json:"_environmentId"`
	OrganizationID string                 `json:"_organizationId"`
	ProviderID     string                 `json:"providerId"`	
	...
}

```